File: solutions/c/darts/1/darts.c
#include "darts.h"

#define OUTER_RADIUS_SQ 100
#define MID_RADIUS_SQ 25
#define INNER_RADIUS_SQ 1

uint8_t score(coordinate_t landing_position)
{
  float x = landing_position[0];
  float y = landing_position[1];
  float sum_x_sq_y_sq = x * x + y * y;

  // check if target missed
  if (sum_x_sq_y_sq > OUTER_RADIUS_SQ)
    return 0;

  // check if dart hits inner circle
  if (sum_x_sq_y_sq <= INNER_RADIUS_SQ)
    return 10;

  // check if dart hits middle circle
  if (sum_x_sq_y_sq <= MID_RADIUS_SQ)
    return 5;

  // dart hit inside the outer circle
  return 1;
}

File: solutions/go/darts/1/darts.go
package darts

func Score(x, y float64) int {
	/**
	 * For a 0,0 centered circle
	 * x^2 + y^2 < r^2 = inside
	 * x^2 + y^2 == r^2 = on
	 * x^2 + y^2 > r^2 = outside
	**/
	dartLoc := x*x + y*y
	outerRadiusSqrd := 100.0
	middleRadiusSqrd := 25.0
	innerRadiusSqrd := 1.0

	if dartLoc > outerRadiusSqrd {
		return 0
	} else if dartLoc > middleRadiusSqrd {
		return 1
	} else if dartLoc > innerRadiusSqrd {
		return 5
	} else {
		return 10
	}

}
